go-controller/pkg/util/kube.go

```diff
@@ -53,13 +55,22 @@ func NewClientset(conf *config.KubernetesConfig) (*kubernetes.Clientset, error)
 		kconfig, err = rest.InClusterConfig()
 	}
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
 	kconfig.AcceptContentTypes = "application/vnd.kubernetes.protobuf,application/json"
 	kconfig.ContentType = "application/vnd.kubernetes.protobuf"
 
-	return kubernetes.NewForConfig(kconfig)
+	kClientset, err := kubernetes.NewForConfig(kconfig)
+	if err != nil {
+		return nil, nil, err
+	}
+	egressFirewallClientset, err := egressfirewallclientset.NewForConfig(kconfig)
```

Comment:
This will not work @JacobTanenbaum : CRD's are not protobuf serializable, you cannot use the `kconfig` on L61-62 for your clientset. You need to initialize it before, see: https://github.com/ovn-org/ovn-kubernetes/pull/1484/commits/da4f352dda6fc463333ded3bc6568fd59d76859b#diff-adcb758b3c8f4b41f256abc916b7eb5fR61

Reply:
done